robot_skills/src/robot_skills/topological_planner.py

```diff
@@ -0,0 +1,68 @@
+# System
+from typing import List
+
+# ROS
+import tf2_ros
+
+# TU/e Robotics
+from topological_action_planner_msgs.srv import Plan, PlanRequest, PlanResponse
+from topological_action_planner_msgs.msg import Edge, Node
+
+# Robot skills
+from .robot_part import RobotPart
+
+
+class TopologicalPlanner(RobotPart):
+    def __init__(self, robot_name: str, tf_buffer: tf2_ros.Buffer):
+        """
+        Interface to the topological planner.
+
+        :param robot_name:
+        :param tf_buffer:
+        """
+        super(TopologicalPlanner, self).__init__(robot_name=robot_name, tf_buffer=tf_buffer)
+
+        # Topological action planner
+        self._planner_srv = self.create_service_client(
+            "/" + self.robot_name + "/topological_action_planner/get_plan",
+            Plan,
+        )
+
+    def get_plan(self, entity_id: str, area: str = None) -> List[Edge]:  # ToDo: add request argument
+        """
+        Gets a plan from the topological action server
+
+        :param entity_id: string identifying the entity
+        :param area: string identifying the area. If this is not defined, the result may be, e.g., a waypoint
+        :return: a list of actions to take
+        """
+        request = PlanRequest()
+        request.origin = Node("", "")  # ToDo: remove!!!
```

Comment:
@LoyVanBeek : this argument (argument) doesn't make any sense. How does the planner handle this?

Reply:
If the origin node is not set, it uses the current location of the robot. If you do specify, then that node is the start of the path, obviously.

Reply:
Okay, then we can remove the line completely?! A `PlanRequest` will have an (empty) origin after initialization anyway?